projects/openapi-io/src/validation/validator.ts

```diff
@@ -99,15 +101,23 @@ export const processValidatorErrors = (
 
   const logger = sourcemap && jsonPointerLogger(sourcemap);
 
-  return pathsWithErrors.map((error) => {
-    const preview = logger
-      ? logger.log(error.instancePath)
-      : `${error.instancePath}`;
-
-    return `${chalk.red('invalid openapi: ')}${chalk.bold.red(
-      getReadableError(error)
-    )}\n${preview}`;
-  });
+  return pathsWithErrors
+    .map((error) => {
+      const pathToLog =
+        error.keyword === 'additionalProperties'
+          ? jsonPointerHelpers.append(
+              error.instancePath,
+              error.params.additionalProperty
+            )
+          : error.instancePath;
+
+      const preview = logger ? logger.log(pathToLog) : `${error.instancePath}`;
+
+      return `${chalk.red('invalid openapi: ')}${chalk.bold.red(
+        getReadableError(error)
+      )}\n${preview}`;
+    })
+    .filter((value, index, array) => array.indexOf(value) === index);
```

Comment:
added this. 

We've seen a few customers get overwhelmed by the number of validation errors. Because we resolve refs we copy them around the spec so 1 error can appear to be 10, but you end up with exactly the same error lines since the sourcemap renders them in regards to their source.